frontend-web/webclient/app/UserSettings/ChangeEmailSettings.tsx

```diff
@@ -0,0 +1,336 @@
+                currentEmailSettings.grantApplicationUpdated = state.settings.grantApplicationUpdated
+                break;
+            case MailType.GRANT_APPLICATION_APPROVED:
+                state.settings.grantApplicationApproved = !state.settings.grantApplicationApproved
+                currentEmailSettings.grantApplicationApproved = state.settings.grantApplicationApproved
+                break;
+            case MailType.GRANT_APPLICATION_REJECTED:
+                state.settings.grantApplicationRejected = !state.settings.grantApplicationRejected
+                currentEmailSettings.grantApplicationRejected = state.settings.grantApplicationRejected
+                break;
+            case MailType.GRANT_APPLICATION_WITHDRAWN:
+                state.settings.grantApplicationWithdrawn = !state.settings.grantApplicationWithdrawn
+                currentEmailSettings.grantApplicationWithdrawn = state.settings.grantApplicationWithdrawn
+                break;
+            case MailType.NEW_COMMENT_ON_APPLICATION:
+                state.settings.newCommentOnApplication = !state.settings.newCommentOnApplication
+                currentEmailSettings.newCommentOnApplication = state.settings.newCommentOnApplication
+                break;
+            case MailType.PROJECT_USER_INVITE:
+                state.settings.projectUserInvite = !state.settings.projectUserInvite
+                currentEmailSettings.projectUserInvite = state.settings.projectUserInvite
+                break;
+            case MailType.PROJECT_USER_REMOVED:
+                state.settings.projectUserRemoved = !state.settings.projectUserRemoved
+                currentEmailSettings.projectUserRemoved = state.settings.projectUserRemoved
+                break;
+            case MailType.VERIFICATION_REMINDER:
+                state.settings.verificationReminder = !state.settings.verificationReminder
+                currentEmailSettings.verificationReminder = state.settings.verificationReminder
+                break;
+            case MailType.USER_ROLE_CHANGE:
+                state.settings.userRoleChange = !state.settings.userRoleChange
+                currentEmailSettings.userRoleChange = state.settings.userRoleChange
+                break;
+            case MailType.USER_LEFT:
+                state.settings.userLeft = !state.settings.userLeft
+                currentEmailSettings.userLeft = state.settings.userLeft
+                break;
+            case MailType.LOW_FUNDS:
+                state.settings.lowFunds = !state.settings.lowFunds
+                currentEmailSettings.lowFunds = state.settings.lowFunds
+                break;
+            case MailType.APPLICATION_STATUS_CHANGE:
+                state.settings.applicationStatusChange = !state.settings.applicationStatusChange
+                currentEmailSettings.applicationStatusChange = state.settings.applicationStatusChange
+                break;
```

Comment:
You're supposed to call the reducer with the new value. I'm not sure how this works in its current state.

Reply:
Is a reducer even needed for this use case? It seems like something which is simple enough that it could just be implemented with a `useState` hook. I am not sure I see the clear benefits of using a custom reducer here.

Reply:
If an object is used instead of a primitive value, I think the reducer is the best approach.